From 590333e675ad0ef51351ac94d08ba3929047673e Mon Sep 17 00:00:00 2001
From: Mohit Sabharwal <mohit@cloudera.com>
Date: Thu, 7 Aug 2014 19:02:06 -0700
Subject: [PATCH 0012/1164] CDH-20686: Forward port parquet-hive-bundle
 classes from CDH-16381

(cherry picked from commit b6df686f670bce17b83e6bc4fa26540da48e07f1)
(cherry picked from commit 69803ff3a43d5f85a3182e9aa6eef0da9c93cb69)
---
 .../parquet/hive/DeprecatedParquetInputFormat.java |   37 ++++++++++++++++++++
 .../hive/DeprecatedParquetOutputFormat.java        |   36 +++++++++++++++++++
 .../parquet/hive/MapredParquetInputFormat.java     |   36 +++++++++++++++++++
 .../parquet/hive/MapredParquetOutputFormat.java    |   35 ++++++++++++++++++
 .../java/parquet/hive/serde/ParquetHiveSerDe.java  |   25 +++++++++++++
 5 files changed, 169 insertions(+)
 create mode 100644 ql/src/java/parquet/hive/DeprecatedParquetInputFormat.java
 create mode 100644 ql/src/java/parquet/hive/DeprecatedParquetOutputFormat.java
 create mode 100644 ql/src/java/parquet/hive/MapredParquetInputFormat.java
 create mode 100644 ql/src/java/parquet/hive/MapredParquetOutputFormat.java
 create mode 100644 ql/src/java/parquet/hive/serde/ParquetHiveSerDe.java

diff --git a/ql/src/java/parquet/hive/DeprecatedParquetInputFormat.java b/ql/src/java/parquet/hive/DeprecatedParquetInputFormat.java
new file mode 100644
index 0000000..ec0ebc0
--- /dev/null
+++ b/ql/src/java/parquet/hive/DeprecatedParquetInputFormat.java
@@ -0,0 +1,37 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package parquet.hive;
+
+import org.apache.hadoop.hive.ql.io.parquet.MapredParquetInputFormat;
+import org.apache.hadoop.io.ArrayWritable;
+
+import parquet.hadoop.ParquetInputFormat;
+
+/**
+ * Deprecated name of the parquet-hive input format. This class exists
+ * simply to provide backwards compatibility with users who specified
+ * this name in the Hive metastore. All users should now use
+ * STORED AS PARQUET
+ */
+@Deprecated
+public class DeprecatedParquetInputFormat extends MapredParquetInputFormat {
+
+  public DeprecatedParquetInputFormat() {
+    super();
+  }
+
+  public DeprecatedParquetInputFormat(final ParquetInputFormat<ArrayWritable> realInputFormat) {
+    super(realInputFormat);
+  }
+}
diff --git a/ql/src/java/parquet/hive/DeprecatedParquetOutputFormat.java b/ql/src/java/parquet/hive/DeprecatedParquetOutputFormat.java
new file mode 100644
index 0000000..a0bdd75
--- /dev/null
+++ b/ql/src/java/parquet/hive/DeprecatedParquetOutputFormat.java
@@ -0,0 +1,36 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package parquet.hive;
+
+import org.apache.hadoop.hive.ql.io.parquet.MapredParquetOutputFormat;
+import org.apache.hadoop.io.ArrayWritable;
+import org.apache.hadoop.mapreduce.OutputFormat;
+
+/**
+ * Deprecated name of the parquet-hive output format. This class exists
+ * simply to provide backwards compatibility with users who specified
+ * this name in the Hive metastore. All users should now use
+ * STORED AS PARQUET
+ */
+@Deprecated
+public class DeprecatedParquetOutputFormat extends MapredParquetOutputFormat {
+
+  public DeprecatedParquetOutputFormat() {
+    super();
+  }
+
+  public DeprecatedParquetOutputFormat(final OutputFormat<Void, ArrayWritable> mapreduceOutputFormat) {
+    super(mapreduceOutputFormat);
+  }
+}
diff --git a/ql/src/java/parquet/hive/MapredParquetInputFormat.java b/ql/src/java/parquet/hive/MapredParquetInputFormat.java
new file mode 100644
index 0000000..9b3d453
--- /dev/null
+++ b/ql/src/java/parquet/hive/MapredParquetInputFormat.java
@@ -0,0 +1,36 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package parquet.hive;
+
+import org.apache.hadoop.io.ArrayWritable;
+
+import parquet.hadoop.ParquetInputFormat;
+
+/**
+ * Deprecated name of the parquet-hive input format. This class exists
+ * simply to provide backwards compatibility with users who specified
+ * this name in the Hive metastore. All users should now use
+ * STORED AS PARQUET
+ */
+@Deprecated
+public class MapredParquetInputFormat extends org.apache.hadoop.hive.ql.io.parquet.MapredParquetInputFormat {
+
+  public MapredParquetInputFormat() {
+    super();
+  }
+
+  public MapredParquetInputFormat(final ParquetInputFormat<ArrayWritable> realInputFormat) {
+   super(realInputFormat);
+  }
+}
diff --git a/ql/src/java/parquet/hive/MapredParquetOutputFormat.java b/ql/src/java/parquet/hive/MapredParquetOutputFormat.java
new file mode 100644
index 0000000..dc6ea3e
--- /dev/null
+++ b/ql/src/java/parquet/hive/MapredParquetOutputFormat.java
@@ -0,0 +1,35 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package parquet.hive;
+
+import org.apache.hadoop.io.ArrayWritable;
+import org.apache.hadoop.mapreduce.OutputFormat;
+
+/**
+ * Deprecated name of the parquet-hive output format. This class exists
+ * simply to provide backwards compatibility with users who specified
+ * this name in the Hive metastore. All users should now use
+ * STORED AS PARQUET
+ */
+@Deprecated
+public class MapredParquetOutputFormat extends org.apache.hadoop.hive.ql.io.parquet.MapredParquetOutputFormat {
+
+  public MapredParquetOutputFormat () {
+    super();
+  }
+
+  public MapredParquetOutputFormat(final OutputFormat<Void, ArrayWritable> mapreduceOutputFormat) {
+    super(mapreduceOutputFormat);
+  }
+}
diff --git a/ql/src/java/parquet/hive/serde/ParquetHiveSerDe.java b/ql/src/java/parquet/hive/serde/ParquetHiveSerDe.java
new file mode 100644
index 0000000..a1dcaa4
--- /dev/null
+++ b/ql/src/java/parquet/hive/serde/ParquetHiveSerDe.java
@@ -0,0 +1,25 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package parquet.hive.serde;
+
+/**
+ * Deprecated name of the parquet-hive output format. This class exists
+ * simply to provide backwards compatibility with users who specified
+ * this name in the Hive metastore. All users should now use
+ * STORED AS PARQUET
+ */
+@Deprecated
+public class ParquetHiveSerDe extends org.apache.hadoop.hive.ql.io.parquet.serde.ParquetHiveSerDe {
+
+}
-- 
1.7.9.5

